src/app/pages/Swap/modals/SwapSelectAsset/EvmChainAssetsList.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { memo, useMemo, MouseEvent, useCallback } from 'react';
+
+import clsx from 'clsx';
+import { FixedSizeList as List, ListChildComponentProps } from 'react-window';
+
+import { EmptyState } from 'app/atoms/EmptyState';
+import { PageLoader } from 'app/atoms/Loader';
+import { DeadEndBoundaryError } from 'app/ErrorBoundary';
+import { useLifiEvmTokensSlugs } from 'app/pages/Swap/modals/SwapSelectAsset/hooks';
+import { useLifiEvmTokensMetadataRecordSelector } from 'app/store/evm/swap-lifi-metadata/selectors';
+import { useEvmTokensMetadataRecordSelector } from 'app/store/evm/tokens-metadata/selectors';
+import { EvmTokenListItem } from 'app/templates/TokenListItem';
+import { EVM_TOKEN_SLUG } from 'lib/assets/defaults';
+import { useEnabledEvmChainAccountTokenSlugs } from 'lib/assets/hooks';
+import { searchEvmChainTokensWithNoMeta } from 'lib/assets/search.utils';
+import { useEvmAccountTokensSortPredicate } from 'lib/assets/use-sorting';
+import { toChainAssetSlug } from 'lib/assets/utils';
+import { useMemoWithCompare } from 'lib/ui/hooks';
+import { useEvmChainByChainId } from 'temple/front/chains';
+import { EvmNetworkEssentials } from 'temple/networks';
+import { TempleChainKind } from 'temple/types';
+
+interface Props {
+  chainId: number;
+  filterZeroBalances: boolean;
+  publicKeyHash: HexString;
+  searchValue: string;
+  onAssetSelect: (e: MouseEvent, chainSlug: string) => void;
+}
+
+interface ItemData {
+  searchedSlugs: string[];
+  publicKeyHash: HexString;
+  network: EvmNetworkEssentials;
+  onAssetSelect: (e: React.MouseEvent, slug: string) => void;
+  chainId: number;
+}
+
+const ITEM_HEIGHT = 56;
+
+export const EvmChainAssetsList = memo<Props>(
+  ({ chainId, filterZeroBalances, publicKeyHash, searchValue, onAssetSelect }) => {
+    const network = useEvmChainByChainId(chainId);
+    if (!network) throw new DeadEndBoundaryError();
+
+    const { lifiTokenSlugs, isLoading } = useLifiEvmTokensSlugs(chainId);
+
+    const tokensSlugs = useEnabledEvmChainAccountTokenSlugs(publicKeyHash, chainId);
+    const tokensSortPredicate = useEvmAccountTokensSortPredicate(publicKeyHash);
+
+    const evmChainAssetsSlugs = useMemo(() => {
+      const gasTokensSlugs: string[] = [EVM_TOKEN_SLUG];
+
+      return gasTokensSlugs.concat(filterZeroBalances ? tokensSlugs : [...tokensSlugs, ...lifiTokenSlugs]);
+    }, [filterZeroBalances, lifiTokenSlugs, tokensSlugs]);
+
+    const evmChainAssetsSlugsSorted = useMemoWithCompare(
+      () => [...evmChainAssetsSlugs].sort(tokensSortPredicate),
```

Comment:
```suggestion
      return gasTokensSlugs.concat(filterZeroBalances ? tokensSlugs : tokensSlugs.concat(lifiTokenSlugs));
    }, [filterZeroBalances, lifiTokenSlugs, tokensSlugs]);

    const evmChainAssetsSlugsSorted = useMemoWithCompare(
      () => evmChainAssetsSlugs.toSorted(tokensSortPredicate),
```
Please avoid spreads wherever possible because they are computationally expensive